### PR TITLE
Improve Cloud Explorer refresh behavior

### PR DESCRIPTION
**Pull request by atwoosnam**
- add "Refreshing..." text indicator
- parallelize all refresh tasks
- start refresh tasks using ThreadingService (for mockability in
  unit tests).
- fetch fresh cf info in background threads, update tree view items
  on UI thread.
- moved Refresh methods from CloudExplorerViewModel into each
  specific ViewModel.
- split refresh-all method on CloudExplorerViewModel into an internal
  async task ("InitiateFullRefresh") and a synchronous caller method
  ("RefreshAllItems") which just invokes
  threadingService.StartTask(InitiateFullRefresh), allowing unit tests
  to await the internal task while the real implementation does not.
- added `expanded` ctor param to TreeViewItemViewModels
- improve unit tests for each TreeViewItemViewModel's specific
  "refresh" method; ensure that expansion doesn't change in the case
  of cfs/orgs/spaces + ensure all states update for apps.
- ensure RefreshAllItems method only invokes underlying task if
  IsRefreshingAll property is false.

### CODE REVIEW

**Comment by atwoosnam** [comment]
Dearest reviewers (@ccheetham @dtillman @zzori-theoriginal),

This PR doesn't include the automatic refresh feature we've been talking about; it's a preliminary set of changes that (A) improve the functionality of the refresh command by parallelizing refresh tasks, (B) cleans up the structure of the code by moving refresh logic out of the main CloudExplorerViewModel & into the individual CF view models (orgs/spaces, etc.), and (C) enhances the unit tests for the whole refresh flow in general (e.g. adding tests for specific corner cases like "already refreshing", adding new mocks that allow the tests to make sure certain tasks were created, adding complexity to prior unit tests, etc.).

The automatic-refresh feature is still a work in progress, but it should be much more approachable now after these changes.

Some of these changes might not be totally intuitive so I've tried to add clarifying comments here & there (especially in the unit tests) -- please reach out if anything still looks odd though! I bet certain parts could still be made clearer / more succinct.

File: src/ViewModels/test/CfInstanceViewModelTests.cs [hunk 1]
```diff
@@ -25,6 +27,14 @@ public void TestInit()
             {
                 _receivedEvents.Add(e.PropertyName);
             };
+
+            MockUiDispatcherService.Setup(mock => mock.
+                RunOnUiThread(It.IsAny<Action>()))
+                    .Callback<Action>(method =>
```
Comment (ccheetham) [inline src/ViewModels/test/CfInstanceViewModelTests.cs:33]: nit:  "action" seems more appropriate then "method"

**Review by ccheetham — APPROVED** [review]
one minor nit/question about var name.  It could well be your name is better.

**Review by zzori-theoriginal — DISMISSED** [review]

File: src/ViewModels/test/CfInstanceViewModelTests.cs (hunk 1 above)
Comment (atwoosnam) [reply src/ViewModels/test/CfInstanceViewModelTests.cs:33]: Fair - I think "action" is technically more accurate

**Review by ccheetham — APPROVED** [review]
lgtm
